test/libsolidity/ASTJSONTest.cpp

```diff
@@ -135,97 +133,85 @@ TestCase::TestResult ASTJSONTest::run(ostream& _stream, string const& _linePrefi
 		sources[m_sources[i].first] = m_sources[i].second;
 		sourceIndices[m_sources[i].first] = static_cast<unsigned>(i + 1);
 	}
-	c.setSources(sources);
-	c.setEVMVersion(solidity::test::CommonOptions::get().evmVersion());
 
+	bool resultsMatch = true;
 
-	if (!c.compile(CompilerStack::State::Parsed))
+	for (TestVariant& variation: m_variations)
 	{
-		SourceReferenceFormatter formatter(_stream, c, _formatted, false);
-		formatter.printErrorInformation(c.errors());
-		return TestResult::FatalError;
-	}
-
-	bool resultsMatch = runTest(
-		m_expectationParseOnly,
-		m_resultParseOnly,
-		sourceIndices,
-		c,
-		"parseOnly",
-		_stream,
-		_linePrefix,
-		_formatted
-	);
+		c.reset();
+		c.setSources(sources);
+		c.setEVMVersion(solidity::test::CommonOptions::get().evmVersion());
 
-	c.reset();
-	c.setSources(sources);
-	c.setEVMVersion(solidity::test::CommonOptions::get().evmVersion());
-	if (!c.parse())
-	{
-		// Empty Expectations means we expect failure
-		if (m_expectation.empty())
-			return resultsMatch ? TestResult::Success : TestResult::Failure;
+		if (!c.parseAndAnalyze(variation.stopAfter))
+		{
+			// Ignore analysis errors, we only want to export.
```

Comment:
This was a bit confusing until I checked what happens inside `analyze()`. So if the stack just exits in the middle of analysis, we treat it as a failure but if it reaches the end and only leaves some errors in `m_errorReporter` then we assume that these are not fatal and we can still try to run subsequent variants?
```suggestion
			// Ignore non-fatal analysis errors, we only want to export.
```

Reply:
The previous code basically ignored any errors coming from the call to `c.analyse()`. Here I tried to unify it while keeping that logic. So if we fail and we reached a stage after `Parsing` then we can safely assume no fatal parsing errors happened.

Reply:
Change added.